### PR TITLE
Remove PIO as part of S2S model build in CMEPS and use hpc-stack

### LINKED ISSUE #197: Load PIO as an external dependency.
**Description**
PIO is required in CMEPS, but CMEPS builds PIO inline.

**Solution**
Use PIO as an external dependency.

### PR DESCRIPTION
**Pull request by aerorahul**
### Description of changes

This PR:
- loads the module pio/2.5.1 from hpc-stack
- removes the building of Parallel IO within CMEPS

### Specific notes

Issues Fixed (include github issue #):
closes #197 

Are changes expected to change answers?
- bit for bit (yes/no)
- more substantial (yes/no)

Specific changes:
- changes in module files: YES

Testing performed:
- machines:
- details:

Hashes used for testing:

- CMEPS:[bafce54](https://github.com/aerorahul/CMEPS/tree/feature/no-pio)

Co-authored by:

Associated PRs:

[CMEPS PR #16](https://github.com/NOAA-EMC/CMEPS/pull/16)
[#199](https://github.com/ufs-community/ufs-s2s-model/pull/199)

### CODE REVIEW

**Comment by JessicaMeixner-NOAA** [comment]
@arunchawla-NOAA didn't you request that PIO was built internally to CMEPS and not as an external library for ease of code delivery, as @rsdunlapiv mentioned in the issue #197

**Comment by arunchawla-NOAA** [comment]
Yes, we initially planned to have pio built as part of the model, but since pio is going to be used in multiple softwares we will get it installed as a library and have it included in WCOSS and RDHPCS

**Comment by junwang-noaa** [comment]
@aerorahul, we just had an s2s commit, would you please merge with the latest s2s, then we can run RT?

**Comment by binli2337** [comment]
@aerorahul, The regression tests have been run on Hera, Orion and WCOSS Dell 3.  The log files are located at  /scratch1/NCEPDEV/nems/Bin.Li/S2S/update/log_files_PR199.

**Comment by aerorahul** [comment]
> @aerorahul, The regression tests have been run on Hera, Orion and WCOSS Dell 3. The log files are located at /scratch1/NCEPDEV/nems/Bin.Li/S2S/update/log_files_PR199.

@binli2337 
I cannot copy Orion logs.
```
1001 hfe01 ~ $ l /scratch1/NCEPDEV/nems/Bin.Li/S2S/update/log_files_PR199/*_orion.intel.log
total 20032
-rw-r----- 1 Bin.Li nems 6656036 Sep 30 06:33 Compile_orion.intel.log
-rw-r----- 1 Bin.Li nems   34686 Sep 30 06:33 RegressionTests_orion.intel.log
```

**Comment by binli2337** [comment]
@aerorahul Please try again. Thanks!

**Comment by aerorahul** [comment]
> @aerorahul Please try again. Thanks!

that worked. thanks @binli2337 
Logs have been updated.

**Review by binli2337 — APPROVED** [review]

**Review by junwang-noaa — APPROVED** [review]

File: .gitmodules [hunk 1]
```diff
@@ -16,8 +16,8 @@
 	branch = master
 [submodule "CMEPS"]
 	path = CMEPS
-	url = https://github.com/NOAA-EMC/CMEPS.git
-	branch = emc/develop
+	url = https://github.com/aerorahul/CMEPS.git
```
Comment (climbfuji) [inline .gitmodules:19]: Sorry if I missed something, is it intentional to point to @aerorahul's CMEPS repo in the merged version of the code?

Comment (aerorahul) [reply .gitmodules:19]: No. not intentional.

Comment (aerorahul) [reply .gitmodules:19]: There is a PR in CMEPS. https://github.com/NOAA-EMC/CMEPS/pull/16
Are they waiting for that one to merge?
I don't know how this works across repos.

Comment (DeniseWorthen) [reply .gitmodules:19]: sorry I missed that merge. It is ready now.

Comment (climbfuji) [reply .gitmodules:19]: Usually it works the other way round, from bottom (innermost nested submodule) to top. Someone merges cmeps, you update your submodule pointer in your ufs-s2s-model PR to point back to the (now merged) official cmeps repo, then merge ufs-s2s-model.

Comment (aerorahul) [reply .gitmodules:19]: @climbfuji 
What do you want me to do now?
Should I open another PR in ufs-s2s-model with the updated .gitmodules and the submodule pointer?

Comment (binli2337) [reply .gitmodules:19]: @aerorahul Please submit a new PR. Thanks!

Comment (aerorahul) [reply .gitmodules:19]: New PR issued. See https://github.com/ufs-community/ufs-s2s-model/pull/205

**Pull request merged**